File: jacinta-helm/unique-in-order.js
var uniqueInOrder=function(iterable){
  console.log(iterable)
  let result = [];
  let last 

  for (let i = 0; i < iterable.length; i++) {
    if (iterable[i] !== last) {
      result.push( last = iterable[i])
    }
  }
    return result;
}


//P create a function that returns a list of items without any elements with the same value.
//R my function should return a new array with elements that dont repeat each other.
//Example ('AAAABBBCCDAABBB'), ['A','B','C','D','A','B']
//Concerns there are no repeated values in the array.
//Explain I can just create a variable for result and last. Use Push method iterating over my array, the indexes. and comparing to the last element.
// making sure that the Index value doesnt equal the last element. 

File: duke-mcclellan/day-38/unique_in_order.py
##Parameters - sequence
##Return - a list of items without elements of the same value
##next to each other
##Example - unique_in_order([1, 2, 2, 3, 3]) == [1, 2, 3]
##Concern - wrong method
##Explain - I want to iterate over the objects in the array.  In each pass, 
##I want the function to detect an identical value and remove it from the array 


def unique_in_order(sequence):
    
    result = []
    for i, item in enumerate(sequence):
        if i == 0 or item != sequence[i -1]:
            result.append(sequence[i])
    return result